libs/dyn/merge/override.go

```diff
@@ -151,10 +147,7 @@ func overrideMapping(basePath dyn.Path, leftMapping, rightMapping dyn.Mapping, v
 				return dyn.NewMapping(), err
 			}
 
-			err = out.Set(rightPair.Key, newValue)
-			if err != nil {
-				return dyn.NewMapping(), err
-			}
+			out.SetLoc(key, rightPair.Key.Locations(), newValue)
```

Comment:
For both diffs in this file, readability regresses. Maybe you could assign both `key` and `keyloc` together to indicate they're one and the same? As is, the diff has a reference to `key` and `rightPair.Key.Locations()`, which doesn't communicate that they are the same originally.

Reply:
good idea, 3eaa1e64fe390f955e60d8606d2e12661f5ff4e5